File: unity/Plugins/Android/src/main/kotlin/com/vibegrowth/sdk/revenue/RevenueTracker.kt
package com.vibegrowth.sdk.revenue

import com.vibegrowth.sdk.identity.UserIdentityManager
import com.vibegrowth.sdk.network.ApiClient
import org.json.JSONObject
import kotlin.concurrent.thread

class RevenueTracker(
    private val apiClient: ApiClient,
    private val identityManager: UserIdentityManager
) {

    fun trackPurchase(amount: Double, currency: String, productId: String) {
        val event = JSONObject().apply {
            put("revenue_type", "purchase")
            put("amount", amount)
            put("currency", currency)
            put("product_id", productId)
        }
        postRevenue(event)
    }

    fun trackAdRevenue(source: String, revenue: Double, currency: String) {
        val event = JSONObject().apply {
            put("revenue_type", "ad_revenue")
            put("amount", revenue)
            put("currency", currency)
            put("ad_source", source)
        }
        postRevenue(event)
    }

    private fun postRevenue(event: JSONObject) {
        thread {
            try {
                val deviceId = identityManager.getOrCreateDeviceId()
                val userId = identityManager.getUserId()
                apiClient.postRevenue(deviceId, userId, event)
            } catch (e: Exception) {
                // Silently handle network errors for revenue tracking
            }
        }
    }
}

File: unity/Plugins/iOS/Sources/Revenue/RevenueTracker.swift
import Foundation

@objc public class RevenueTracker: NSObject {
    private let apiClient: ApiClient
    private let identityManager: UserIdentityManager

    @objc public init(apiClient: ApiClient, identityManager: UserIdentityManager) {
        self.apiClient = apiClient
        self.identityManager = identityManager
        super.init()
    }

    @objc public func trackPurchase(pricePaid: Double, currency: String, productId: String? = nil) {
        var event: [String: Any] = [
            "revenue_type": "purchase",
            "amount": pricePaid,
            "currency": currency,
        ]
        if let productId {
            event["product_id"] = productId
        }
        postRevenue(event: event)
    }

    @objc public func trackAdRevenue(source: String, revenue: Double, currency: String) {
        let event: [String: Any] = [
            "revenue_type": "ad_revenue",
            "amount": revenue,
            "currency": currency,
            "ad_source": source
        ]
        postRevenue(event: event)
    }

    private func postRevenue(event: [String: Any]) {
        DispatchQueue.global(qos: .background).async { [weak self] in
            guard let self = self else { return }
            let deviceId = self.identityManager.getOrCreateDeviceId()
            let userId = self.identityManager.getUserId()
            self.apiClient.postRevenue(deviceId: deviceId, userId: userId, event: event, completion: nil)
        }
    }
}
